accesscontrol/jwt.go

```diff
@@ -123,12 +123,60 @@ func NewJWTFromJWKS(jwtConf *config.JWT, jwks *jwk.JWKS) (*JWT, error) {
 	}
 
 	algorithms := append(acjwt.RSAAlgorithms, acjwt.ECDSAlgorithms...)
-	jwtAC.parser = newParser(algorithms)
+	var algos []string
+	for _, a := range algorithms {
+		algos = append(algos, a.String())
+	}
+
+	jwtAC.algos = algos
 	jwtAC.jwks = jwks
 
 	return jwtAC, nil
 }
 
+type parserConfig struct {
+	algorithms []string
+	audience   string
+	issuer     string
+}
+
+func (p parserConfig) key() string {
+	return fmt.Sprintf("%s:%s:%s", p.algorithms, p.audience, p.issuer)
+}
+
+func (p parserConfig) newParser() *jwt.Parser {
```

Comment:
I don't think we are concurrency safe here. Could you introduce a `func (j *JWT) Validate(req *http.Request) error` test which runs in parallel ? Using the MemCache with a global/internal key-prefix would to the trick here.